lib/puppet/type.rb

```diff
@@ -1316,6 +1318,7 @@ def pathbuilder
       and the second run will log the edit made by Puppet.)"
 
     validate do |list|
+      Puppet.deprecation_warning(_("The `audit` metaparameter is deprecated and will be ignored in a future release."))
```

Comment:
I agree that this should definitely be at least `warn_once` if it's otherwise per instance in a resource; that sounds insane. Is docs-only enough to get people to remove it?